functions-and-operators: enrich instructions for string functions CHAR_LENGTH() and CHARACTER_LENGTH #15483 (#16057)
pingcap/docs: functions-and-operators/string-functions.md

### [`BIT_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bit-length)

> The `BIT_LENGTH()` function is used to return the length of a given argument in bits. It calculates the length of the argument and then returns the result in bits.

Reviewer: ```suggestion
The `BIT_LENGTH()` function is used to return the length of a given argument in bits.
```

Reviewer: The second sentence seems unnecessary as it expresses the same meaning as the first sentence.

Examples:

```sql
SELECT BIT_LENGTH("TiDB");
+----------+
| BIT_LENGTH("TiDB") |
+----------+
|  32 |
+----------+
```

8 bits per character x 4 characters = 32 bits

```sql
SELECT CustomerName, BIT_LENGTH(CustomerName) AS BitLengthOfName FROM Customers;
| CustomerName|BitLengthOfName |
|-------------|----------------|
| Albert Einstein | 120 |
| Robert Oppenheimer | 144 |
```

```sql

> SELECT BIT_LENGTH("PingCap 123");
> +----------+
> | BIT_LENGTH("PingCap 123") |

Reviewer: ```suggestion
SELECT BIT_LENGTH("PingCAP 123");
+----------+
| BIT_LENGTH("PingCAP 123") |
```

+----------+
| 88 |
+----------+
```

8 bits per character (space is counted as it is a non-alphanumeric character) x 11 characters = 88 bits

> **Note:**
>

> > The second example operates under the assumption that there is a database with a row titled `Customers` and a column inside titled `CustomerName`

Reviewer: ```suggestion
> The second example operates under the assumption that there is a database with a table named `Customers` and a column inside named `CustomerName`.
```

### [`CHAR()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char)

> The `CHAR()` function is used to get the corresponding character value of a specific ASCII value. It performs the opposite function of `ASCII()`. Unlike `ASCII()`, which can get the ASCII value of a specific character, `CHAR()` can get the character of a specific ASCII value.

Reviewer: ```suggestion
The `CHAR()` function is used to get the corresponding character of a specific ASCII value. It performs the opposite operation of `ASCII()`, which returns the ASCII value of a specific character.
```

Examples:

```sql
SELECT CHAR(65);

+------------+
|  CHAR(65) |
+------------+
|        A |
+------------+
```

```sql
SELECT CHAR(84);

+------------+
|  CHAR(84) |
+------------+
|        T |
+------------+
```

> **Note:**
>

> > The `CHAR()` function can also be used to get the corresponding character values of ASCII values that extend beyond the standard ASCII range (0 - 127). It can also get the corresponding character value of a unicode value.

Reviewer: ```suggestion
> The `CHAR()` function can also be used to get the corresponding character of ASCII values that extend beyond the standard ASCII range (0 - 127). It can also get the corresponding character value of a unicode value.
```

Further Examples:

```sql
/*For extended ASCII: */

SELECT CHAR(128);

+------------+
|  CHAR(128) |
+------------+
|       0x80 |
+------------+
```

```sql
/* For Unicode: */

--skip-binary-as-hex

SELECT CHAR(50089);

+------------+
|  CHAR(50089) |
+------------+
|   é |
+------------+
```

### [`CHAR_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char-length)

> The `CHAR_LENGTH()` function is used to get the number of characters in a specified expression. It returns the total number of characters in a specified expression as an integer value. 

Reviewer: ```suggestion
The `CHAR_LENGTH()` function is used to get the total number of characters in a given argument as an integer value.
```

Examples:

```sql
SELECT CHAR_LENGTH("TiDB") AS LengthOfString;

+----------+
| LengthOfString |
+----------+
| 4  |
+----------+
```

```sql
SELECT CustomerName, CHAR_LENGTH(CustomerName) AS LenghtOfName FROM Customers;

| CustomerName|LengthOfName  |
|-------------|--------------|
| Albert Einstein | 15 |
| Robert Oppenheimer | 18 |
```

> **Note:**
>

> > The second example operates under the assumption that there is a database with a table titled `Customers` and a column inside the table titled `CustomerName`

Reviewer: ```suggestion
> The second example operates under the assumption that there is a database with a table named `Customers` and a column inside the table named `CustomerName`.
```

### [`CHAR_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char-length)

> The `CHAR_LENGTH()` function is used to get the number of characters in a specified expression. It returns the total number of strings in a specified expression as an integer value. 

Reviewer 2: ```suggestion
The `CHAR_LENGTH()` function is used to get the number of characters in a specified expression. It returns the total number of characters in a specified expression as an integer value. 
```

Examples:

```sql
SELECT CHAR_LENGTH("TiDB") AS LengthOfString;

+----------+
| LengthOfString |
+----------+
| 4  |
+----------+
```

```sql
SELECT CustomerName, CHAR_LENGTH(CustomerName) AS LenghtOfName FROM Customers;

| CustomerName|LengthOfName  |
|-------------|--------------|
| Albert Einstein | 15 |

> | Robert Oppenheimer | 18 |

Reviewer 2: Maybe include a name where `CHAR_LENGTH()` and `LENGTH()` don't match:
```
mysql> WITH n AS (
    -> SELECT 'François Englert' AS p
    -> UNION ALL SELECT 'Albert Einstein'
    -> UNION ALL SELECT 'Robert Oppenheimer'
    -> )
    -> SELECT p, CHAR_LENGTH(p), LENGTH(p) FROM n;
+--------------------+----------------+-----------+
| p                  | CHAR_LENGTH(p) | LENGTH(p) |
+--------------------+----------------+-----------+
| François Englert   |             16 |        17 |
| Albert Einstein    |             15 |        15 |
| Robert Oppenheimer |             18 |        18 |
+--------------------+----------------+-----------+
3 rows in set (0.00 sec)
```

Author: I think this should be introduced in the `LENGTH()` subheading where the difference between `LENGTH()` and `CHAR_LENGTH()` will be stated and this type of example will be included. Adding it here when `LENGTH()` is yet to be introduced to the readers can confuse.

> ```

Reviewer 2: ````suggestion
```

````

> **Note:**
>

> > The second example operates under the assumption that there is a MySQL database with a record titled `Customers` and a field inside titled `CustomerName`

Reviewer 2: ```suggestion
> The second example operates under the assumption that there is a database with a record titled `Customers` and a field inside titled `CustomerName`
```

### [`CHAR_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char-length)

The `CHAR_LENGTH()` function is used to get the number of characters in a specified expression. It returns the total number of characters in a specified expression as an integer value. 

Examples:

```sql
SELECT CHAR_LENGTH("TiDB") AS LengthOfString;

+----------+
| LengthOfString |
+----------+
| 4  |
+----------+
```

```sql
SELECT CustomerName, CHAR_LENGTH(CustomerName) AS LenghtOfName FROM Customers;

| CustomerName|LengthOfName  |
|-------------|--------------|
| Albert Einstein | 15 |
| Robert Oppenheimer | 18 |
```

> **Note:**
>

> > The second example operates under the assumption that there is a database with a record titled `Customers` and a field inside titled `CustomerName`

Reviewer 3: Not sure if the word "record" is proper, in my understanding, it's a table named "Customers" with a column named "CustomerName"

Author: changed to row and column to ensure consistency

### [`CHAR_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char-length)

The `CHAR_LENGTH()` function is used to get the total number of characters in a given argument as an integer.

Examples:

```sql
SELECT CHAR_LENGTH("TiDB") AS LengthOfString;

+----------------+
| LengthOfString |
+----------------+
|              4 |
+----------------+
```

```sql
SELECT CustomerName, CHAR_LENGTH(CustomerName) AS LenghtOfName FROM Customers;

+--------------------+--------------+
| CustomerName       | LenghtOfName |
+--------------------+--------------+
| Albert Einstein    |           15 |
| Robert Oppenheimer |           18 |
+--------------------+--------------+
```

> **Note:**
>

> > The preceding example operates under the assumption that there is a database with a table named `Customers` and a column inside the table named `CustomerName`.

Reviewer 4: ```suggestion
> The preceding examples operate under the assumption that there is a database with a table named `Customers` and a column inside the table named `CustomerName`.
```

Reviewer: only this example
![image](https://github.com/pingcap/docs/assets/79440533/7f1b2ce0-63f7-4d81-87f6-c5c52c714b02)

### [`CHARACTER_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_character-length)

> The `CHARACTER_LENGTH()` function is the same as the `CHAR_LENGTH()` function. Both functions can be used synonymously because they provide the same outputs.

Reviewer 4: ```suggestion
The `CHARACTER_LENGTH()` function is the same as the `CHAR_LENGTH()` function. Both functions can be used synonymously because they generate the same outputs.
```
